### ISSUE TITLE
Copy to clipboard actions not working

### ISSUE THREAD

**Issue by beelze** [Robbendebiene/Gesturefy#252]
I'm using FF58.0.1 on Linux. I've tried Copy tab URL, Copy link URL and Copy text selection actions but nothing was copied. Maybe it is a linux-specific issue?

**Comment by Robbendebiene** [comment]
Hey, thanks for the report. After some tests on Linux and Windows this seems to be a Linux sepcific bug, which is already [reported to mozilla](https://bugzilla.mozilla.org/show_bug.cgi?id=1418665).

**Comment by beelze** [comment]
There is some possible delay workaround, as stated [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1418665#c2). Can you provide me an instructions to test it?

**Comment by Itchiii** [comment]
Hey together. 
I tried a few solutions, but without success. I list the approaches so you can better understand them.
All attempts have worked on Windows, not on Linux.

Or maybe I miss something or you see an error / improvement?.

- added a delay before copying the text 
```  javaScript
/* lib/actions.js */
  CopyTextSelection: function (data) {
    let input = document.createElement("textarea");
    document.body.append(input);
    input.value = data.textSelection;
    input.select();
    setTimeout(function functionName() {
      document.execCommand('Copy');
      //document.body.removeChild(input); or even that with removing the last statement
    },200);
    document.body.removeChild(input);
  },
```
- added a delay as a wrapper like [here (bugzilla reporter)](https://github.com/S2-/pwgen/blob/master/src/popup/pwgen.js#L51)
```  javaScript
/* lib/actions.js */
  CopyTextSelection: function (data) {
    setTimeout(function functionName() {
      let input = document.createElement("textarea");
      document.body.append(input);
      input.value = data.textSelection;
      input.select();
      document.execCommand('Copy');
      document.body.removeChild(input);
    },200);
  },
```
- changed the delay time (until 2000ms)

- added own background page in manifest.json ->[delevoper.mozilla](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/manifest.json/background)
``` JavaScript
/*manifest.json*/
  "background": {
    //remove "scripts" and put in the html page
    "page": "my-background.html"
  },
```
``` html
<!--my-background.html-->
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <script type="text/javascript" src="lib/actions.js"></script>
    <script type="text/javascript" src="lib/backgroundCommons.js"></script>
    <script type="text/javascript" src="lib/background.js"></script>
  </head>
  <body>

  </body>
</html>
```
I hope you have more ideas. ^^

**Comment by Robbendebiene** [comment]
@Itchiii Thank you very much for your effort.

Unfortunately the reason why the timeout is working in other extensions is because they execute the copy command in a content script or popup and not in the background script (what I have to do). After digging through a lot of bugs addressing clipboard problems I found [this one](https://bugzilla.mozilla.org/show_bug.cgi?id=1272869).

The reason why it's working on windows is, because Firefox on windows runs webextensions out-of-process which is not enabled by default on Linux or macOS. You can turn it on in about:config by setting `extensions.webextensions.remote` to true, however this will surely have some side effects.

**Comment by beelze** [comment]
Thanks, `extensions.webextensions.remote` helped to fix this issue. Seems it worth to put some notification about this option for Linux users (if WE API allows to identify the platform) somewhere around clipboard actions on addon' conf page.

**Comment by Robbendebiene** [comment]
@beelze I agree, therefore I also will keep the issue open until there is a real fix available.

**Comment by Robbendebiene** [comment]
Fixed since [Gesturefy 2.0.4](https://github.com/Robbendebiene/Gesturefy/releases/tag/v2.0.4)
